src/Arcus.Scripting.IntegrationAccount/Scripts/Set-AzIntegrationAccountAgreements.ps1

```diff
@@ -0,0 +1,87 @@
+Param(
+    [Parameter(Mandatory = $true)][string] $ResourceGroupName = $(throw "Resource group name is required"),
+    [Parameter(Mandatory = $true)][string] $Name = $(throw "Name of the Integration Account is required"),
+    [parameter(Mandatory = $false)][string] $AgreementFilePath = $(if ($AgreementsFolder -eq '') { throw "Either the file path of a specific agreement or the file path of a folder containing multiple agreements is required, e.g.: -AgreementFilePath 'C:\Agreements\agreement.json' or -AgreementsFolder 'C:\Agreements'" }),
+    [parameter(Mandatory = $false)][string] $AgreementsFolder = $(if ($AgreementFilePath -eq '') { throw "Either the file path of a specific agreement or the file path of a folder containing multiple agreements is required, e.g.: -AgreementFilePath 'C:\Agreements\agreement.json' or -AgreementsFolder 'C:\Agreements'" }),
+    [Parameter(Mandatory = $false)][string] $ArtifactsPrefix = ''
+)
+
+if ($AgreementFilePath -ne '' -and $AgreementsFolder -ne '') {
+    throw "Either the file path of a specific agreement or the file path of a folder containing multiple agreements is required, e.g.: -AgreementFilePath 'C:\Agreements\agreement.json' or -AgreementsFolder 'C:\Agreements'"
+}
+
+function UploadAgreement {
+    param(
+        [Parameter(Mandatory = $true)][System.IO.FileInfo] $Agreement
```

Comment:
Maybe rename this to `$ArgumentFilePath` or something, having it like this reminds me of the actual model.

Reply:
That would also mean that `UploadAgreement -Agreement $agreement` will become `UploadAgreement -ArgumentFilePath $agreement`. I personally like the Agreement naming a bit better as it is more descriptive.
A change in this would also mean changing it in all Set-AzIntegrationAccount*.ps1 files (already doing some changes for this PR though 😉 )

Reply:
Ah, I see. It's more an item in the directory. Hmm, yeah, we can leave it like this then. It would otherwise be ambiguous with the already defined `$AgreementFilePath` in the script params.